### PR TITLE
Plots

### LINKED ISSUE #9: additional plots
>        any ideas about some other plots for this repo - bar charts, etc?

_Originally posted by @shcheklein in https://github.com/iterative/vscode-dvc-demo/issues/7#issuecomment-1319087177_

### PR DESCRIPTION
**Pull request by dberenbaum**
Adds more plots. Closes #9.

### CODE REVIEW

**Review by mattseddon — APPROVED** [review]

File: train.py [hunk 1]
```diff
@@ -143,10 +151,15 @@ def main():
         train(model, x_train, y_train, params["lr"], params["weight_decay"])
         torch.save(model.state_dict(), "model.pt")
         # Evaluate and checkpoint.
-        metrics, actual, predicted = evaluate(model, x_test, y_test)
-        for k, v in metrics.items():
-            live.log_metric(k, v)
+        metrics_train, _, _ = evaluate(model, x_train, y_train)
+        for k, v in metrics_train.items():
+            live.log_metric(f"{k}_train", v)
```
Comment (daavoo) [inline train.py:156]: I would recommend:

```suggestion
            live.log_metric(f"train/{k}", v)
```

IMO, it look nicer in the tree view and the table

**Review by shcheklein — APPROVED** [review]

File: dvc.yaml [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
   - training/plots/images
   - training/plots/sklearn/confusion_matrix.json:
       template: confusion
       x: actual
       y: predicted
+  - training/plots/sklearn/confusion_matrix.json:
+      template: jitter.json
+      x: actual
+      y: predicted
+      title: Actual vs Predicted
```
Comment (mattseddon) [inline dvc.yaml:23]: @dberenbaum it looks like you can only use the dataset for one plot/template:

image

Version of DVC used: `2.34.2`

Comment (mattseddon) [reply dvc.yaml:23]: The jitter template also seems to tank the extension's performance. Is there any way that we can change the template and aggregate the data before it is output by `plots diff`? (it also causes so less noticeable issues in the HTML shown above)

Comment (shcheklein) [reply dvc.yaml:23]: > The jitter template also seems to tank the extension's performance. 

Do we know the reason for this? 

Confusion matrix is still needed, yep.

Comment (mattseddon) [reply dvc.yaml:23]: > Do we know the reason for this?

I assume the reason is that we are not pre-processing the raw data. That means we are passing 10k+ data points for each revision. We are also rendering all of those points in a single plot.

Chrome even struggles with this plot:

https://user-images.githubusercontent.com/37993418/202976513-e851451c-9123-4dc7-ba29-1e61846a01b9.mov

Comment (dberenbaum) [reply dvc.yaml]: > it looks like you can only use the dataset for one plot/template

Fixed.

> we are plotting 10k+ data points

Downsampled to 500 points, although no guarantee users will do the same.

File: train.py (hunk 1 above)
Comment (dberenbaum) [reply train.py]: Made this change, but there were a couple reasons I used the other syntax initially:
1. `train/{k}` probably breaks on Windows.
2. I think it's not that clear whether the header in `train/acc.tsv` should be `acc` or `train/acc` (right now, it's `train/acc`).

File: dvc.yaml (hunk 2 above)
Comment (shcheklein) [reply dvc.yaml]: @dberenbaum thanks! should bring the confusion matrix back?

Comment (dberenbaum) [reply dvc.yaml]: Yup, everything should be there. The problem was having the same ID for different plots. Making a note to provide a better warning/error for that.

Screenshot 2022-11-21 at 3 30 33 PM

**Pull request merged**